### DISCUSSION TITLE
Preprocess lines before superfences

### DISCUSSION THREAD

**Discussion by pawamoy** [facelessuser/pymdown-extensions#2163] [Q&A] [answered]
Hello, I'm writing a small Markdown extension that detects pycon code blocks without indentation or fences.

```md
Following is detected as pycon code block:

>>> 1 + 1
2
```

When it finds such blocks, it wraps them in fences for SuperFences to handle them. The above block becomes:

````md
```pycon
>>> 1 + 1
2
```
````

Initially I tried writing the extension as a block processor (it was implemented in one line!) but it looks like SuperFences runs as a preprocessor, so I had to re-implement it as a preprocessor as well, to run before SuperFences.

My preprocessing works, but only when the block is not indented (i.e. not in an admonition or tab), and only when the block is not already in a code block.

````md
This does not work:

```

>>> 1 + 1
2
```

This neither:

!!! note

    >>> 1 + 1
    2
````

This is very limited.

How would you go about writing such an extension :thinking:? Is there a way with a block processor instead of a preprocessor?

**Comment by pawamoy** [comment, marked as answer]
OK, once again I was able to use @oprypin's code :sweat_smile: 
I reused the highlighter from mkdocstrings in a block processor :slightly_smiling_face:

**Reply by facelessuser** [reply]
I assume you solved your issue?

**Reply by pawamoy** [reply]
Yes!

**Reply by facelessuser** [reply]
Perfect
